### PR TITLE
Reload interval

### PR DESCRIPTION
**Pull request by phunkyfish**
v4.6.0
- Added: M3U auto refresh
- Update: update github links for Matrix
- Update: Remove dependency on p8-platform util, thread and mutex
- Added: Switch EPGTimeshift setting to hours instead of minutes

### CODE REVIEW

File: src/PVRIptvData.h [hunk 1]
```diff
@@ -34,6 +31,9 @@
 
 #include <atomic>
 
+#include <kodi/libXBMC_pvr.h>
```
Comment (ksooo) [inline src/PVRIptvData.h:36]: Other addons do not consider "kodi" as system headers. Why this change?

File: src/PVRIptvData.cpp [hunk 2]
```diff
@@ -57,10 +58,27 @@ bool PVRIptvData::Start()
 
 void PVRIptvData::Process()
 {
+  unsigned int refreshTimer = 0;
+  time_t lastRefreshTimeSeconds = time(nullptr);
```
Comment (ksooo) [inline src/PVRIptvData.cpp]: std::time

File: src/PVRIptvData.cpp [hunk 3]
```diff
@@ -57,10 +58,27 @@ bool PVRIptvData::Start()
...
+  int lastRefreshHour = Settings::GetInstance().GetM3URefreshHour(); //ignore if we start during same hour
+
   while (m_running)
   {
     std::this_thread::sleep_for(std::chrono::milliseconds(PROCESS_LOOP_WAIT_SECS * 1000));
 
+    time_t currentRefreshTimeSeconds = time(nullptr);
```
Comment (ksooo) [inline src/PVRIptvData.cpp]: std:.time

File: src/PVRIptvData.h (hunk 1 above)
Comment (phunkyfish) [reply src/PVRIptvData.h:36]: I found that some kodi headers used system <> and some used regular "".

It made sense to me that as these are not referenced relatively to the local source, i.e. relative to an include path they should be referenced as system headers. Do you think that reasoning is incorrect?

Comment (ksooo) [reply src/PVRIptvData.h:36]: I think your reasoning is correct. Maybe you should contact @AlwinEsch and ask him to change that for all official binary addons?

**Review by ksooo — APPROVED** [review]

File: src/PVRIptvData.h (hunk 1 above)
Comment (phunkyfish) [reply src/PVRIptvData.h:36]: Will do
